src/settings/DepartmentsSettings.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {
+  FormattedMessage,
+  useIntl,
+} from 'react-intl';
+
+import { NoValue } from '@folio/stripes/components';
+import { ControlledVocab } from '@folio/stripes/smart-components';
+import { useStripes } from '@folio/stripes/core';
+
+const DepartmentsSettings = () => {
+  const intl = useIntl();
+  const stripes = useStripes();
+  const ConnectedControlledVocab = stripes.connect(ControlledVocab);
+
+  const validate = (item, index, items) => {
+    const filteredDepartments = items.filter((department, i) => i !== index);
+    const errors = {};
+
+    if (filteredDepartments.find(department => department.name === item.name)) {
+      errors.name = <FormattedMessage id="ui-users.settings.departments.name.error" />;
+    }
+
+    if (filteredDepartments.find(department => department.code === item.code)) {
+      errors.code = <FormattedMessage id="ui-users.settings.departments.code.error" />;
+    }
+
+    return errors;
+  };
+
+  return (
+    <ConnectedControlledVocab
+      stripes={stripes}
+      baseUrl="departments"
+      records="departments"
+      label={intl.formatMessage({ id: 'ui-users.settings.departments' })}
+      labelSingular={intl.formatMessage({ id: 'ui-users.settings.department' })}
+      objectLabel={<FormattedMessage id="ui-users.settings.departments.users" />}
+      visibleFields={['name', 'code']}
+      columnMapping={{
+        name: intl.formatMessage({ id: 'ui-users.settings.departments.name' }),
+        code: intl.formatMessage({ id: 'ui-users.settings.departments.code' }),
```

Comment:
Nice work!
As recommendation, you can extract `formatMessage` above to make code cleaner:
`const { formatMessage } = useIntl();`